### DISCUSSION TITLE
@click.away behavior

### DISCUSSION THREAD

**Discussion by nuernbergerA** [alpinejs/alpine#807] [1. Help] [answered]
```html
<div x-data="{ open: false }">
    <button @click="open = true">Open Modal</button>

    <div
        x-show="open"
        @click.away="open = false"
    >
        <textarea></textarea>
    </div>
</div>
```
Hey people,

if I select the text inside the `textarea` and release the mouse button (`onmouseup`) outside of the modal, it closes the model.

Wouldn't it be better if `@click.away` listens for `onmousedown`?

I use `@mousedown.away` to "fix" the behavior, but somehow that doesn't feel right to me.

Did anyone else share my expectation?

Cheers Adrian

**Comment by SimoTod** [comment, marked as answer]
Please see the official specs https://developer.mozilla.org/en-US/docs/Web/API/Element/click_event
```
If the button is pressed on one element and the pointer is moved outside the element before the button is released, the event is fired on the most specific ancestor element that contained both elements.
```
Alpine does not define events, it uses native ones. I hope the above link helps.

**Reply by nuernbergerA** [reply]
Hey Simone,

i know that this if the default JS behavior and not an alpine issue.
Was just wondering if people encounter this problem too and how they handle it.

Maybe the example should use `@mousedown.away` because I think this is what people really want

**Reply by SimoTod** [reply]
Yeah my observation was because you suggested click should listen for mousedown which is not possible because Alpine doesn't redefine events and it will be against the spec. Usually people don't move their mouse while single clicking if they don't have a good reason and click is semantically more meaningful then mousedown so the example doesn't seem bad to me but it's just my personal opinion 👍
